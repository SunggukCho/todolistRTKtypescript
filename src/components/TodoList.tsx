import { Todo } from '../types/todo' 

const TodoList = (props: any) => {
  return (
    <>
      {props.todos.map((todo: Todo) => (
        <div key={todo.id}>
          <input type="checkbox" onChange={() => props.toggle(todo)} checked={todo.done} />
          {todo.id} {todo.value} <button onClick={()=>props.handleDelete(todo)}>-DEL</button>
        </div>
      ))}
    </>
  )
}

export default TodoList