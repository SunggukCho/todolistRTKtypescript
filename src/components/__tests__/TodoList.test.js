import { render, screen } from '@testing-library/react';
import TodoList from '../TodoList';

const TodoListProps = {
  todos: [
    {
      "id": 1,
      "value": "Hello",
      "done": true
    },
    {
      "id": 2,
      "value": "World!",
      "done": false
    }
  ],
  handleSubmit: () => {
    console.log('handle submit test')
  },
  handleDelete: (deleteTodo) => console.log('handle submit test delete'),
  toggle: (checkedTodo) => console.log('handle submit test toggle')
}

test("TodoList 엘리먼트에 `Todo`라는 글자가 포함되어있는가?", ()=>{
  // 텍스트 render 체크
  // TodoList 엘리먼트에 "Todo"라는 글자가 포함되어있는가?
  render(<TodoList {...TodoListProps} />)
  const TodoListEl = screen.getByText(/Todo/i)
  expect(TodoListEl).toBeInTheDocument();
})

test("TodoList 스냅샷 체크", () => {
  // 스냅샷 활용
  // 미리찍어둔 화면과 렌더링된 화면이 같은지를 체크함.
  // 처음 테스트를 하면 snapshots라는 디렉토리가 생성됨.
  const el = render(<TodoList {...TodoListProps} />)
  expect(el).toMatchSnapshot();
})

