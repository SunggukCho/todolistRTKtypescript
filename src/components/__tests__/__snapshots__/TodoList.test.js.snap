// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList 스냅샷 체크 1`] = `
<div>
  <h2>
    Todo
  </h2>
  <div>
    <input
      checked=""
      type="checkbox"
    />
    1
     
    Hello
     
    <button>
      -DEL
    </button>
  </div>
  <div>
    <input
      type="checkbox"
    />
    2
     
    World!
     
    <button>
      -DEL
    </button>
  </div>
</div>
`;
